src/api/mod.rs

```diff
@@ -23,8 +23,37 @@ pub mod api_client;
 pub mod error;
 pub mod rpc_api;
 
+use ac_node_api::EventDetails;
 use serde::{Deserialize, Serialize};
 
+/// Extrinsic report returned upon a submit_and_watch request.
+/// Holds as much information as available.
+#[derive(Debug, Clone)]
+pub struct ExtrinsicReport<Hash> {
+	// Hash of the extrinsic.
+	pub extrinsic_hash: Hash,
+	// Block hash of the block the extrinsic was included in.
+	// Only available if watched until at least `InBlock`.
+	pub block_hash: Option<Hash>,
+	// Last known Transaction Status.
+	pub status: TransactionStatus<Hash, Hash>,
+	// Events assosciated to the extrinsic.
+	// Only available if explicitly stated, because
+	// extra node queries are necessary to fetch the events.
+	pub events: Option<Vec<EventDetails>>,
```

Comment:
That's not yet used, but will be soon with the incoming `watch_until_success` function.

Reply:
Do you plan that one can submit explicit events that we should wait for? I ask because, we can't know at this level, which events are associated to an extrinsic. We only know which ones are in this block.

Reply:
Yes, the extrinsics are retrieved from the block, but they can be sorted to `associated events` with the phase:
```rust
// Filter events associated to our extrinsic.
		let associated_event_results = events.iter().filter(|ev| {
			ev.as_ref()
				.map_or(true, |ev| ev.phase() == Phase::ApplyExtrinsic(xt_index as u32))
		});
```
polkadot.js does the same :

![grafik](https://user-images.githubusercontent.com/73821294/209085788-8e610f8a-7f58-4a8b-b7f0-7f1abaf6eac1.png)

Reply:
Oh wow, I did not know about that feature, this is very cool!